core/tests/test_utils.py

```diff
@@ -1926,37 +1870,61 @@ def login(self, email, is_super_admin=False):
             email: str. The email of the user who is to be logged in.
             is_super_admin: bool. Whether the user is a super admin.
         """
-        self.testbed.setup_env(
-            overwrite=True,
-            user_email=email, user_id=self.get_auth_id_from_email(email),
-            user_is_admin=('1' if is_super_admin else '0'))
+        os.environ['USER_ID'] = self.get_auth_id_from_email(email)
+        os.environ['USER_EMAIL'] = email
+        os.environ['USER_IS_ADMIN'] = ('1' if is_super_admin else '0')
 
     def logout(self):
         """Simulates a logout by resetting the environment variables."""
-        self.testbed.setup_env(
-            overwrite=True, user_email='', user_id='', user_is_admin='0')
+        os.environ['USER_ID'] = ''
+        os.environ['USER_EMAIL'] = ''
+        os.environ['USER_IS_ADMIN'] = '0'
 
     @contextlib.contextmanager
-    def mock_datetime_utcnow(self, mocked_datetime):
-        """Mocks response from datetime.datetime.utcnow method.
+    def mock_datetime_utcnow(self, mocked_now):
+        """Mocks parts of the datastore to accept a fake datetime type that
+        always returns the same value for utcnow.
 
-        Example usage:
+        Example:
             import datetime
-            mocked_datetime_utcnow = (
-                datetime.datetime.utcnow() - datetime.timedelta(days=1))
-            with self.mock_datetime_utcnow(mocked_datetime_utcnow):
-                print datetime.datetime.utcnow() # prints time reduced by 1 day
-            print datetime.datetime.utcnow() # prints current time.
+            mocked_now = datetime.datetime.utcnow() - datetime.timedelta(days=1)
+            with mock_datetime_for_datastore(mocked_now):
```

Comment:
Where does mock_datetime_for_datastore come from (where is it defined)?

Reply:
Renamed.